iceoryx_dds/source/gateway/main.cpp

```diff
@@ -32,7 +32,7 @@ class ShutdownManager
   public:
     static void scheduleShutdown(int num)
     {
-        char reason;
+        char reason = '\0';
```

Comment:
Fun fact. gcc states here `maybe uninitialized` but detects this error only when `and_then` / `or_else` has a template functor argument.